[docs] Improve instructions about how to build/use HEAD (#4529)
drud/ddev: docs/content/developers/building-contributing.md

# Building, Testing, and Contributing

## Testing Latest Commits on HEAD

> You have many options for getting and using the latest committed HEAD version of DDEV:

Reviewer: Can be shortened:

```suggestion
There are several ways to use DDEV’s latest-committed HEAD version:
```

> * **Download** the latest artifacts from the master branch from [link](https://nightly.link/drud/ddev/workflows/master-build/master). This version is signed and notarized, built by the CI system. Get the one you need and place it in your `$PATH`.

Reviewer: “link” isn’t descriptive text, but “nightly.link” at least hints at where they’ll live. Reworded for brevity. Used “each of these” to transition from plural (“artifacts”) to singular items instead of jumping straight to “This version”. Reordered for chronological accuracy (I think).

```suggestion
* **Download** the latest master branch artifacts from [nightly.link](https://nightly.link/drud/ddev/workflows/master-build/master). Each of these is built by the CI system, signed, and notarized. Get the one you need and place it in your `$PATH`.
```

> * **Homebrew install HEAD**: On macOS and Linux, you can `brew unlink ddev && brew install drud/ddev/ddev --HEAD --fetch-HEAD` to get the latest commit of DDEV, even if it’s unreleased. Since you're building this on your own computer, it's not signed or notarized, and you'll get a notification that instrumentation doesn't work, which is fine.  If you’re using Linux/WSL2, you’ll likely need to install build-essential with the following command: `sudo apt install -y build-essential`.

Reviewer: “run” commands, extremely minor brevity improvement:

```suggestion
* **Homebrew install HEAD**: On macOS and Linux, run `brew unlink ddev && brew install drud/ddev/ddev --HEAD --fetch-HEAD` to get the latest DDEV commit, even if it’s unreleased. Since you’re building this on your own computer, it’s not signed or notarized, and you’ll get a notification that instrumentation doesn’t work, which is fine. If you’re using Linux/WSL2, you’ll likely need to install build-essential by running the following command: `sudo apt install -y build-essential`.
```

> * **Build manually**: If you have normal build tools like `make` and `go` installed, you can just check out the code and `make`.

Reviewer: Eliminate “just”, and “run” command.

```suggestion
* **Build manually**: If you have normal build tools like `make` and `go` installed, you can check out the code and run `make`.
```
